os/arch/arm/src/armv8-m/up_exception.S

```diff
@@ -165,23 +196,30 @@ exception_common:
 	mov     sp, r1					/* Set the MSP to the PSP */
 
 1:
+
 	mov		r2, sp					/* R2=Copy of the main/process stack pointer */
 	add		r2, #HW_XCPT_SIZE		/* R2=MSP/PSP before the interrupt was taken */
-									/* (ignoring the xPSR[9] alignment bit) */
+
 #ifdef CONFIG_REG_STACK_OVERFLOW_PROTECTION
-	mov		r3, #0x0
+	mov		r0, #0x0
 	ittee		eq
 	mrseq		r1, msplim
-	msreq		msplim, r3
+	msreq		msplim, r0
 	mrsne		r1, psplim
-	msrne		psplim, r3
+	msrne		psplim, r0
 	stmdb		sp!, {r1}
 #endif
-#ifdef CONFIG_ARMV8M_USEBASEPRI
-	mrs		r3, basepri				/* R3=Current BASEPRI setting */
-#else
-	mrs		r3, primask				/* R3=Current PRIMASK setting */
-#endif
+
+	/* If we are entering in handler mode, and pervious mode was thread mode with psp,
```

Comment:
Please fix the typo : `and the previous`

Reply:
Done